### DISCUSSION TITLE
What kernel version do you need for multishot timeouts?

### DISCUSSION THREAD

**Discussion by cmazakas** [axboe/liburing#959] [Q&A] [answered]
I'm reading the man page for `io_uring_prep_timeout.3` and I'm noticing that I keep getting `-EINVAL` when I setup the multishot timeout SQE.

I looked into liburing's test and it seems like it _skips_ this if the platform doesn't support it. I went into `test/timeout.c` and updated the code to do:
```c
ret = cqe->res;
if (ret == -EINVAL) {
  printf("noooooooo, say it ain't soooo!!!!!!\n");
	no_multishot = 1;
	return T_EXIT_SKIP;
}
```
and in my terminal, I see:
```
exbigboss@exbigboss-XPS-9315:~/cpp/liburing$ ./test/timeout.t 
noooooooo, say it ain't soooo!!!!!!
```

So it seems like this is expected on my platform. I'm using Ubuntu 23.04 w/ kernel 6.2 so I'm a little surprised this doesn't seem to work:
```
exbigboss@exbigboss-XPS-9315:~/cpp/liburing$ uname -a
Linux exbigboss-XPS-9315 6.2.0-33-generic #33-Ubuntu SMP PREEMPT_DYNAMIC Tue Sep  5 14:49:19 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by cmazakas** [comment]
Alright, I upgraded to Ubuntu 23.10 from 23.04 (i.e. 6.2 -> 6.5) and now it seems to be working.

**Comment by axboe** [comment, marked as answer]
It was added with 6.4 - usually I add that to the man pages, but looks like this one was forgotten. Will do that now.
